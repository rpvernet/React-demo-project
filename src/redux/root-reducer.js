import { combineReducers} from "redux";
import {persistReducer} from "redux-persist";
import storage from 'redux-persist/lib/storage'

import userReducer from "./user/user.reducer";
import postReducer from "./post/post.reducer";
import replyReducer from "./reply/reply.reducer"


const persistConfig = {
    key:'root',
    storage

}

const rootReducer = combineReducers({
    user:userReducer,
    posts:postReducer,
    reply:replyReducer
});

export default persistReducer(persistConfig, rootReducer);