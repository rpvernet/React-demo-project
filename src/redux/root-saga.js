import {all, call} from 'redux-saga/effects';

import {userSagas} from "./user/user.saga";
import {postSagas} from "./post/post.sagas";
import {replySagas} from "./reply/reply.sagas";


export default function* rootSaga(){
    yield all([
        call(userSagas),
        call(postSagas),
        call(replySagas)

    ])

}