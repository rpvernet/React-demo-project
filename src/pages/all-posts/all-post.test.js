import React from 'react';
import { Provider} from "react-redux";
import { combineReducers, createStore } from 'redux';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store' //ES6 modules


import AllPosts from './all-posts.component';

/*describe('All posts', () =>{
    it('render All posts', () =>{
        const props = {
            useDispatch:jest.fn(),
            isLoading:false,
            currentUser:{
                id:'12324',
                displayName:'tasty test'
            },
            sortType:'newestFirst',
            posts:{
                id:'222',
                postTitle:"let's test"
            }
        }

        const wrapper = mount(<Provider store={props}>
            <AllPosts {...props}/>
        </Provider>)
        expect(wrapper).toMatchSnapshot();
        })
    })*/



export const createMockStore = ({ state, reducers }) => {
     const store = createStore(combineReducers(reducers), state);
     return {
         ...store,
         persistor: {
             persist: () => null
         }
     };
};


 describe ('UserPosts', () => {
     let store;
     let wrapper;
     let mockFetchPostsStart;

     beforeEach(() => {
         const mockPostReducer = (
             state = {
                 posts:{isFetching: true
                 }
             }, action) => state;

        const mockUserReducer = (
            state = {
                user:{
                    currentUser:{
                        id:'1232',
                        displayName:'Tasty testy'
                    }
            }
            }, action)=> state;




         const mockState = {
             posts: {
                 isFetching: true
             },
             user:{
               currentUser:{
                   id:'1232',
                   displayName:'Tasty testy'
               }
             }
         };

         mockFetchPostsStart = jest.fn();

         store = combineReducers({mockUserReducer, mockPostReducer})




         const mockProps = {
             fetchPostsStart: mockFetchPostsStart
         };

         wrapper =shallow(
             <Provider store={store}>
                 <AllPosts {...mockProps}/>
             </Provider>
           );
     })


     it('should render AllPosts component', () => {
         expect(wrapper).toMatchSnapshot();
     });

});
