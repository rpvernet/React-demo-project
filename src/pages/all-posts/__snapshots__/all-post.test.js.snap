// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPosts should render AllPosts component 1`] = `ShallowWrapper {}`;
